### ISSUE TITLE
Go wild with intellisense?

### ISSUE THREAD

**Issue by lars-erik** [marcemarc/tooorangey.XPathOfLeastResistance#3]
This is cool.
I'm raising you several units tho.
Got this malnourished pet:
https://github.com/lars-erik/our-umbraco-query-language

Call, fold or ignore. Now you know. ;)

**Comment by marcemarc** [comment]
@lars-erik yes I have seen your ambitious project :-P and it is interesting...

but with this, and it's next up in terms of my spare time to look again, the aim is the sort of reverse of your query language - I don't want to abstract querying, so people can use a simple query language and be sure they are using the best performant implementation underneath (although that is a fantastic aim in itself to have) - instead I'm just trying to, by example, illustrate some simple common XPath expressions, in the context of someone's Umbraco site, so people can feel ok about using XPath if they feel a little unsure about it. Have also noticed people who do use XPath tend to go for //NewsArticle, when in lots of cases the more direct XPath /root/News/newsArticle would be more performant...

Essentially selfishly, we spend alot of time taking over unperforming Umbraco sites, using in memory querying, and it's become a routine switch to TypeContentAtXPath, and it's a pain to guess the capitalisations of the alaises...

... so I think I just need some more examples, some text and a query tester? in the slideout panel... and then it can just pend...

until we need the 'Umbraco Query Language' Edition :-) to help people get their heads around what you are building in the context of their Umbraco site!

**Comment by lars-erik** [comment]
Sounds good.

I can't help dreaming about an XPath input box with IntelliSense suggesting the "most performant" query your current text matches the best. >< 
I.E. "newsArt" yields autofill like ["/root/News/newsArticle (58)", "//newsArticle (67)", ...] and showing count + execution time whenever you blur or click preview.
Dunno how much complex AI vs. a few simple rules would be needed, but would be sweet.

Then again, I wholeheartedly respect the matter of time, ambition and actual current requirements/needs. 😆
